### DISCUSSION TITLE
Is there a way to connect using connection strings/ability to use flags while connecting

### DISCUSSION THREAD

**Discussion by shivamstaq** [ra1u/redis-dart#46] [Q&A]
Hey,
I am trying to use Redis deployed in the cloud. I could not find a way to supply the password that comes with the deployment. 
Either a way to be able to connect using username/password that may be active, or just the ability to connect using a redis:// connection string. If it's not there, are there any plans to add it? 
Thanks

**Comment by ra1u** [comment]
Hi!

Have you tried using command [AUTH](https://redis.io/commands/auth)

I am not sure, following will work as intended as I dont have test environment here, but you can start with something like. 
    
    final conn =  RedisConnection();
    final command = await conn.connect('addr', port);
    final rsp = await command.send_object(["AUTH","username","password"]);
    if(rsp is RedisError){
        print(rsp);
    }

Let me know if this works for you or  how you see this library can be improved for this.

Make sure that 'addr' is address (something you can ping) not uri staring with redis:// . In case of uri, you should parse string and extract address and port depending on your [syntax](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier#Syntax)

**Reply by shivamstaq** [reply]
Thanks for the response. 
The above solution does not work. The following is the error that I get on the conn.connect() command:
`Unhandled Exception: SocketException: OS Error: Connection timed out, errno = 110, address = db-redis-###############.b.db.ondigitalocean.com, port = 42028`

(also, 42028 is not the port number I am passing to the connect method. )
I see that there is a way to connect using a Socket object, but even that does not seem to have a way to connect using authentication params or connection strings. 

The cli command I use for connection is:
`redis-cli -h host -p port -a password`
We can pass host and port to the connection, but not the -a flag (or any other flags). 

The library [`dartis`](https://pub.dev/packages/dartis) does offer connection using connection strings, but it has not been able to roll out a null-safety update yet. I have not been able to test if the connection works with `dartis` because under the hood it also seems to be parsing out only host and port from the `redis://` connection string.
The library [web_socket_channel](https://pub.dev/packages/web_socket_channel) is what flutter suggests ([here](https://pub.dev/packages/web_socket_channel)) for socket connections using connection strings. Not sure if it would work for `redis://` too but worth a shot.

**Reply by ra1u** [reply]
Interesting.

What conn.connect() does is it just opens socket connection on address and port and sends no data and also receives no data. From your report it seems that error is somewhere else. 

Do you run TLS socket on server? In such case, you should open TLS connect with conn.connectSecure in similar matter.

**Reply by shivamstaq** [reply]
connectSecure also gives the same error. I figured it would be because of the authentication.
What else can be the problem ?

**Reply by shivamstaq** [reply]
I think since [`redis-dart`](https://github.com/ra1u/redis-dart) relies on `darti:io` Socket for connection, the ability to connect using a connection string/ authentication params will need to be implemented there before it can be added to this project. 

I have added an issue on `dart:io` [here](https://github.com/dart-lang/io/issues/82)
